public/projects/today_i_learned_cli.md

```diff
@@ -0,0 +1,268 @@
+# Today I Learned - Command Line
+
+During your programming journey you will come across questions every day and need to find answers for those questions. Recent studies have found a [cognitive benefit to journaling](https://psychcentral.com/lib/the-health-benefits-of-journaling/) when learning new content, or balancing between multiple priorities. For many people the act of writing down a problem, the path to finding a solution, and the answer, stimulates pathways in the brain which help in memory recall and problem solving.
+
+Your task for this project is to build a command line program which you can use to record, search, edit, and reference answers for various programming questions and answers.
+
+## Wire Frames
+
+<!-- Maybe wire frames are not going to be helpful for the CLI version -->
+[TBD]()
+
+## Considerations
+
+* How should you ensure each TIL entry should has a unique name?
+* How should you record the Date and Time of each TIL entry?
+* How should you 'tag' a TIL entry with a category type?
+* How could you find and edit an existing TIL entry?
+* How could you write long TIL entries?
+
+## User Stories
+
+### Creating a new TIL entry
+
+#### Given ####
+An empty command prompt
+
+#### When ####
+
+A user inputs:
+
+``` sh
+til -n convert-json-to-javascript -t javascript -c "To convert JSON to a Javascript object use JSON.parse(myObject)"
```

Comment:
I think this is too many options for a first feature. Let's add categories later (default to 'general') and I don't see the value in naming entries; a TIL journal is just a list of notes, and it should be as frictionless as possible e.g. `til all numbers in JS are floats` 💥